### PR TITLE
Add proper types for Badge and a bunch of the theme code

### PR DESCRIPTION
**Pull request by Rowno**
This adds proper TypeScript types for all the Badge components and a bunch of the theme code as part of the TypeScript migration.

### CODE REVIEW

File: src/typography/src/Text.tsx [hunk 1]
```diff
@@ -10,17 +10,17 @@ export interface TextProps extends React.ComponentProps<typeof Box> {
   /**
    * The color (alias or valid color) applied to the text
    */
-  color?: string
+  color: string
 
   /**
    * The font family alias applied to the text
    */
-  fontFamily?: FontFamily
```
Comment (mshwery) [inline src/typography/src/Text.tsx]: These aren't really required though – are they?

File: src/typography/src/Paragraph.tsx [hunk 2]
```diff
@@ -10,19 +10,19 @@ export interface ParagraphProps extends React.ComponentProps<typeof Box> {
...
    * The color (alias or valid color) applied to the text
    */
-  color?: string
+  color: string
 
   /**
    * Font family.
    * Can be: `ui`, `display` or `mono` or a custom font family.
    */
-  fontFamily?: FontFamily
```
Comment (mshwery) [inline src/typography/src/Paragraph.tsx]: These aren't really required are they? Can't you pass `undefined` to them and still get back the right values?

File: src/theme/src/ThemeContext.ts [hunk 3]
```diff
@@ -7,53 +7,32 @@ export interface Theme {
...
-  getIconSizeForIconButton: (height: number) => number
-  getLinkClassName: (color?: string) => string
-  getParagraphStyle: (size?: number) => any
-  getTagInputClassName: (alias: 'default') => string
-  getTextColor: (colorAlias?: string) => string
-  getTextDropdownButtonClassName: () => string
-  getTextInputClassName: (alias: 'none') => string
-  getTextSizeForControlHeight: (height: number) => number
-  getTextStyle: (size?: number) => any
+  getAlertProps: typeof defaultTheme.getAlertProps
```
Comment (mshwery) [inline src/theme/src/ThemeContext.ts]: :awyis:

File: src/badges/src/Badge.tsx [hunk 4]
```diff
@@ -2,24 +2,36 @@ import React, { PureComponent } from 'react'
...
+  color: string
 
-    /**
-     * Whether or not to apply hover/focus/active styles
-     */
-    isInteractive: PropTypes.bool,
+  /**
+   * Whether or not to apply hover/focus/active styles
+   */
+  isInteractive: boolean
```
Comment (mshwery) [inline src/badges/src/Badge.tsx]: Technically this shouldn't be required?

File: src/typography/src/Text.tsx (hunk 1 above)
Comment (Rowno) [reply src/typography/src/Text.tsx]: They are, but they have `defaultProps` so you don't need to set them.

File: src/typography/src/Paragraph.tsx (hunk 2 above)
Comment (Rowno) [reply src/typography/src/Paragraph.tsx]: Some of the theme functions don't handle undefined cases. But these have `defaultProps` so you don't need to set them.

File: src/badges/src/Badge.tsx (hunk 4 above)
Comment (Rowno) [reply src/badges/src/Badge.tsx]: It has a `defaultProp`.

**Pull request closed without merging**